File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        std::sort(strs.begin(), strs.end());
        string lcp = "";
        string first = strs[0];
        string last = strs[strs.size() - 1];

        for (int i = 0; i < min(first.length(), last.length()); i++){
            if (first[i] != last[i]){
                return lcp;
            }
            lcp += first[i];
            }
        return lcp;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        lcp = ""
        strs.sort()
        first = strs[0]
        last = strs[-1]
        for letter in range(min(len(first), len(last))):
            if first[letter] != last[letter]:
                return lcp
            lcp += first[letter]
        return lcp